### DISCUSSION TITLE
Pre-Built Packages

### DISCUSSION THREAD

**Discussion by dsandros** [devpi/devpi#950] [Q&A]
Is there a way to upload pre-built packages?

I am trying to implement devpi in an more unique use case. We do not have it setup to be a PyPi mirror and have no internet connection. Once a package is approved for use in our environment we want to go out and pull the package and upload it pre-built if possible. We currently upload the package on a machine with pypi connectivity then essentially copy all of devpi from that machine and move it to our isolated server.

**Comment by fschulze** [comment]
You can use ``twine`` or ``devpi upload`` to upload wheels and source releases. Is that what you mean?
